conductr_cli/logging_setup.py

```diff
@@ -36,6 +55,37 @@ def screen(self, message, *args, **kwargs):
     self.log(LOG_LEVEL_SCREEN, message, *args, **kwargs)
 
 
+def progress(self, message, *args, **kwargs):
+    """
+    This method expects `flush` keyword argument.
+
+    If `flush` is set to `False`, then the terminal cursor is moved to the beginning of the line after message is
+    printed. This allows text replacement within the terminal, and is used to render something like a progress bar.
+
+    If `flush` is set to `True`, then line break character is appended to the message, causing the terminal cursor to
+    move to the next line.
+
+    Thus, the correct use of the progress logging are:
+
+    ```
+    log.progress('1%', flush=False) # Terminal shows 1%
+    ...
+    log.progress('99%', flush=False) # Terminal shows 99% replacing whatever previous value is
+    log.progress('Done', flush=True) # Terminal replaces whatever previous value with `Done`, moving to new line
+
+    # continue as normal
+    log.info('Hey')
+    ```
+    """
+    flush_required = kwargs.pop('flush')
+
+    line_end = '\r'
+    if flush_required:
+        line_end = '\n'
```

Comment:
@ktoso - replaced multiple backspace character with `\r`; much better - thank you!